src/Orleans.Core.Abstractions/GrainFeature.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Orleans
+{
+    public class GrainInterfaceMetadata
+    {
+        public GrainInterfaceMetadata(Type interfaceType, Type referenceType, Type invokerType, int interfaceId)
+        {
+            this.InterfaceType = interfaceType;
+            this.ReferenceType = referenceType;
+            this.InvokerType = invokerType;
+            this.InterfaceId = interfaceId;
+        }
+
+        public Type InterfaceType { get; }
+
+        public Type ReferenceType { get; }
+
+        public Type InvokerType { get; }
```

Comment:
Maybe I confused GrainClassXXX with Invoker stuff. This InvokerType is not necessarily available on the client, right? But the rest of this information must be available on the client, correct?
If that's the case, shouldn't InvokerType be part of `GrainClassMetadata`?
Or what is this type? Some XML comments on all of these would help a lot

Reply:
Invoker *is* available on the client, since it's associated with the interface and not the class. We don't generate any code for the class (until now, with metadata)